src/test/java/com/klarna/hiverunner/examples/InsertTestData.java

```diff
@@ -79,6 +78,20 @@ public void insertRowsFromTsvFile() {
 
 
     @Test
+    public void insertRowsFromTsvFileWithHeader() {
+        File dataFile = new File("src/test/resources/examples/dataWithHeader1.tsv");
+        TsvFileParser parser = new TsvFileParser().withHeader(true);
+        String[] header = parser.parseHeader(dataFile);
+        shell.insertInto("source_db", "test_table")
```

Comment:
That was my initial idea but I couldn't figure out how to do it in a nice way as the parsing of the file and the insert into is a bit separated in how they work together. I can revisit it though. 

Anyway, thanks for the comments. I'll revisit my pull request and make some of your suggested edits.